lib/LaTeXML/Core/Mouth.pm

```diff
@@ -288,49 +284,53 @@ sub readToken {
       $$self{lineno}++;
       $$self{colno} = 0;
       my $line = $self->getNextLine;
+      # For \read, we have to return something for EOL, and handle implicit final newline
+      my $read_mode = (($STATE->lookupValue('PRESERVE_NEWLINES') || 0) > 1);
+      my $eolch     = "\r";
+      if (my $eol = $STATE->lookupDefinition(T_CS('\endlinechar'))) {
+        $eol   = $eol->valueOf()->valueOf;
+        $eolch = (($eol > 0) && ($eol <= 255) ? chr($eol) : undef); }
       if (!defined $line) {    # Exhausted the input.
+        my $eolcc    = ((defined $eolch) && $STATE->lookupCatcode($eolch)) // CC_OTHER;
```

Comment:
@brucemiller a bit tangential, but it would be more robust to have `lookupCatcode` return `CC_OTHER` by default, like TeX does. Right now you add `// CC_OTHER` to every call to `lookupCode`, I think this would become the fifth line, see e.g. [this patch](https://github.com/brucemiller/LaTeXML/compare/master...xworld21:b66a301).